documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,182 @@
+
+An example of a verifiable credential with a `credentialStatus` of type `RevocationBitmap2022`.
+
+```json
+{
+  "@context": "https://www.w3.org/2018/credentials/v1",
+  "id": "https://example.edu/credentials/3732",
+  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+  "credentialSubject": {
+    "id": "did:iota:B8DucnzULJ9E8cmaReYoePU2b7UKE9WKxyEVov8tQA7H",
+    "GPA": "4.0",
+    "degree": "Bachelor of Science and Arts",
+    "name": "Alice"
+  },
+  "issuer": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw",
+  "issuanceDate": "2022-06-13T08:04:36Z",
+  "credentialStatus": {
+    "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#revocation",
+    "type": "RevocationBitmap2022",
+    "revocationBitmapIndex": "5"
+  },
+  "proof": {
+    "type": "JcsEd25519Signature2020",
+    "verificationMethod": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#key-1",
+    "signatureValue": "2eHdbDumMrer4pNVkaiYMqsVqVp2adq7bRcgTJZiw17Zeghk2ZT49YHwLwCCg35YKganBhxP6YSbzYoBK1AuCUv"
+  }
+}
+```
+
+and the corresponding issuer's DID document where credential `"5"` in the `#revocation` service is revoked:
+
+```json
+{
+  "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw",
+  "verificationMethod": [
+    {
+      "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#key-1",
+      "controller": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw",
+      "type": "Ed25519VerificationKey2018",
+      "publicKeyMultibase": "z3hgM9fNkhwgT5mECbj1HdKoFNZgpffwQYEV8WBVHphXq"
+    }
+  ],
+  "capabilityInvocation": [
+    {
+      "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#sign-0",
+      "controller": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw",
+      "type": "Ed25519VerificationKey2018",
+      "publicKeyMultibase": "z83F6zbD3KqaxvQhqo25LvSXzoDdpZmp3EpPVonSVACwZ"
+    }
+  ],
+  "service": [
+    {
+      "id": "did:iota:EvaQhPXXsJsGgxSXGhZGMCvTt63KuAFtaGThx6a5nSpw#revocation",
+      "type": "RevocationBitmap2022",
+      "serviceEndpoint": "data:application/octet-stream;base64,ZUp5ek1tQmdZR1NBQUFFZ1ptVUFBQWZPQUlF"
+    }
+  ]
+}
+```
+
+## Test Vectors
+
+This section provides test vectors to validate implementations against.
+
+### Test Vector 1
+
+The following data url decodes to a bitmap of length 0 where no index is revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQUFBd0FES0FCcg=="`
+
+### Test Vector 2
+
+The following data url decodes to a bitmap of length 3 where indices `5`, `398`, and `67000` are revoked:
+
+`"data:application/octet-stream;base64,ZUp5ek1tQmdZR0lBQVVZZ1pHQ1FBR0laSUdabDZHUGN3UW9BRXVvQjlB"`.
+
+### Test Vector 3
+
+The following data url decodes to a bitmap of length 16384 where all indices are revoked:
+
+`"data:application/octet-stream;base64,ZUp6dHhERVJBQ0FNQkxESEFWS1lXZkN2Q3E0MmFESmtyMlNrM0ROckFLQ2RBQUFBQUFBQTMzbGhHZm9q"`
+
+## Rationale
+
+This section describes the rationale behind some of the design decisions of this specification.
```

Comment:
This could probably be expanded with comparisons to `RevocationList2020` and why some aspects of `RevocationBitmap2022` make it preferable, and the limitations thereof (maximum size).

Reply:
Expanded that section, please take a look. Would be nice to give precise numbers on how many revocations are possilble with the Tangle message size limits, but I think it's pretty hard to calculate, since (I think) the roaring bitmap's serialization size doesn't scale linearly with the number of contained elements?

Reply:
Its size scales with the number of integer blocks required to represent set bits, conceptually. That's why setting random bits approximates its worst case size, which is true for compression too. I'll take a look later.

Reply:
See https://github.com/iotaledger/identity.rs/pull/879#discussion_r900161631